### ISSUE TITLE
Connection of 6 PWM-channels

### ISSUE THREAD

**Issue by pluschris** [bigdisp/scoreboard-hardware#16]
@gn36: I found a uC with 6 PWM channels. When there are only two 7-segment-digits connected we can control the color of each digit separately. But if there are additionally two dots, I've no chance to control that color separately, too.

Do you prefer to control
- the color of the digits and the color of the dots separately or
- the color of the digits separately and a dot has the same color as on digit?

**Comment by gn36** [comment]
Can't we solve the problem by doing the pwm in software? That is what I did so far at least. See [interrupt.c](https://github.com/bigdisp/scoreboard-atmega8-i2c/blob/master/src/interrupt.c#L124) and [pwm.c](https://github.com/bigdisp/scoreboard-atmega8-i2c/blob/master/src/pwm.c#L8).

Then it would be primarily a question of a sufficient number of output pins and software timers, not whether they support pwm or not. I haven't tried this with multiple rgb outputs so far, though. 

Getting back to your original question anyways: I think we need more than a single layout. Just for illustration, consider the old draft of the board:
![g5615](https://cloud.githubusercontent.com/assets/607501/14941652/9cd7591a-0fa3-11e6-9e29-bd3c83016456.png)
- The display of the score ("Gast", "Heim") consists of two digits for each team and those two obviously belong together. So their color can be controlled together. Unless we recycle the point on these numbers for the point that marks which team is currently batting (which I wouldn't do, actually, because it logically belongs to the display of the inning and there should be enough channels on that one), I don't see any use for the point at all on these digits, so we can omit it entirely. Thus, we could use the 6 pwm channels for controlling the two digits separately here because we don't need the dot. 
- The display of the inning needs 1.5 digits, but two dots. Originally, I was considering using the dots from the score ("Gast", "Heim") for this, but really, the 1.5 digits is only a single port (if we wire the point to the one in front), so we have the entire second port for the two dots. Here, I would like different colors for the dots and the number. If we can manage that, maybe even two different colors for the two dots, but that is not strictly necessary. If we simply put the two dots on the other port, we can use the same board layout as for the score digits, except that most of the port that is used for the first digit on "Heim" and "Gast" is not connected for "Inning". Wiring the point output on "Heim" and "Gast" to the same pwm ports, even though it is not used there doesn't really hurt either.
- The special digit for Balls, Strikes, and Outs is more difficult. Ideally, I would like a different setting for all three, because they all provide different information and it would make it easier to distinguish. Fortunately, we will only need a single port for controlling the segments, it is essentially a 7 segment display without dot. So we could use your 6 pwm ports for two of the three and realize the third pwm in software using the second port. Unfortunately, that means we cannot recycle the hardware for this and this would need special software (which I would integrate and switch using settings in the eeprom again).

So in conclusion, except for the special digit for "Balls", "Strikes", and "Outs", we should do separate colors on the two digits and consider the dot being part of the digit to be colored in the same color. The special digit needs a different layout capable of separating the elements of one digits into different colors.

**Comment by pluschris** [comment]
Obviously different software versions and different boards aren't that nice concerning maintaining later on. 
So I thought about it and it the following would cover all your requirements:
- Full control to the main controller=Raspberry (every strip can be addressed from the main controller, no mapping at the uC. That makes new figures and effects pretty easy: You don't need to open the uC-case...)
- 4uC-boards (each has 2x3 PWM-channels: 1x3 PWMs for two digits and 1x3 PWMs for two dots). That makes sure that brightness does not depend on other interrupt activities), for example:
  - "Gast"-number and inning dot 1
  - "Heim"-number and inning dot 2
  - Inning-number and outs
  - Balls and strikes

You could control the color of each block individually and we need just one board and one software.

**Comment by gn36** [comment]
Ok that works.
